lib/src/top_level.rs

```diff
@@ -561,9 +610,17 @@ fn fn_to_doc<'a>(
     args: &'a Vec<(String, CoqType)>,
     ret_ty: &'a Option<CoqType>,
     body: &'a Expr,
+    is_dead_code: bool,
 ) -> Doc<'a> {
+    let mut dead_code_attribute = String::new();
+    if is_dead_code {
+        dead_code_attribute
+            .push_str("(* #[allow(dead_code)] - function was ignored by the compiler *)");
+        dead_code_attribute.push('\n');
+    }
     nest([
```

Comment:
Wrapping everything into a `group` construct to align the comment with the beginning of the definition:
```suggestion
    group([
      if is_dead_code {
        concat([
          text("(* #[allow(dead_code)] - function was ignored by the compiler *)"),
          hardline(),
        ])
      } else {
        nil()
      },
      nest([
```